File: LAMP/Week5/bike.php
<?php
class Bike {
    public $price;
    public $max_speed;
    public $miles;

    public function __construct($dollars, $mph) {
        $this->price = $dollars;
        $this->max_speed = $mph;
        $this->miles = 0;
    }
    public function displayInfo() {
        echo 'Price:'.$this->price.'<br />';
        echo 'Max Speed:'.$this->max_speed.'<br />';
        echo 'Miles Driven: '.$this->miles.'<br />';
    }
    public function drive() {
        echo 'Driving...<br />';
        $this->miles += 10;
        return $this;
    }
    public function reverse() {
        echo 'Reversing...<br />';
        $this->miles -= 5;
        if ($this->miles < 0) {
            $this->miles = 0;
        }
        return $this;
    }
}

$bike1 = new Bike(100, 85);
$bike1->drive()->drive()->drive()->reverse()->displayInfo();

$bike2 = new Bike(400, 55);
$bike2->drive()->drive()->reverse()->reverse()->displayInfo();

$bike3 = new Bike(1200, 77);
$bike3->reverse()->reverse()->reverse()->displayInfo();
 ?>

File: OOP/bike/bike.py
import random
class Bike(object):
    def __init__(self, price, max_speed):
        self.price = price
        self.max_speed = max_speed
        self.miles = 0

    def displayInfo(self):
        print "The price is", str(self.price)
        print "The max speed is", str(self.max_speed) + 'mph'
        print "The miles on that bike is", str(self.miles) + ' miles'

    def rides(self):
        print "Riding"
        self.miles += 10
        return self

    def reverse(self):
        print "Reversing"
        if self.miles >= 5:
            self.miles -= 5
        return self

bike1 = Bike(300, 28)
bike1.rides().rides().rides().reverse().displayInfo()

bike2 = Bike(300, 10)
bike2.rides().rides().reverse().reverse().displayInfo()
